### ISSUE TITLE
Gson TypeAdapters :: No json name generated for `@Nullable` list/array

### ISSUE THREAD

**Issue by almozavr** [immutables/immutables#359]
### Context
Model contains nullable array (or List):
```java
@Nullable
@SerializedName("tags")
String[] tags();
```
### Problem
Generated TypeAdapter doesn't set name for array serialization:
```java
@Nullable String[] tagsElements = instance.tags();
if (tagsElements != null) {
    // <---- missing out.name("tags");
    out.beginArray();
    for (String e : tagsElements) {
      out.value(e);
    }
    out.endArray();
} else if (out.getSerializeNulls()) {
    out.name("tags");
    out.nullValue();
}
```

**Comment by elucash** [comment]
Could it be the same as #357 ?

**Comment by elucash** [comment]
Ok, probably not, have to take a look, thanks!

**Comment by elucash** [comment]
Released in 2.2.6, Thank you!

**Comment by almozavr** [comment]
Great, thank you!!

Чт, 9 черв. 2016 21:43 користувач Eugene <notifications@github.com> пише:
